### DISCUSSION TITLE
Use Entity Graph with projections

### DISCUSSION THREAD

**Discussion by z8manu8z** [Cosium/spring-data-jpa-entity-graph#169] [Q&A] [answered]
Hello all,

I try to use Cosium Entity Graph library with JPA projections (https://docs.spring.io/spring-data/jpa/reference/repositories/projections.html) but I get an InapplicableEntityGraphException.
It seems that the only type possible for repositories methods is model class (defined in EntityGraphJpaRepository<ModelClass, Long>).
Any idea on how to return interfaces ?

Thanks !

**Comment by reda-alaoui** [comment, marked as answer]
Under the hood, `spring-data-jpa` projections lead to multi select JPA queries which are not compatible with JPA EntityGraph. You can't have both.

**Reply by z8manu8z** [reply]
Thanks for your quick reply,

I currently use JPA projections in combination with JPA entity graphs.
For example to retrieve a Product object with or without the list of corresponding Orders.
This works great and allows me to retrieve the entire tree with a single query.
The major disadvantage is that the graphs used are static and therefore do not allow me to have a single method with a parameter allowing me to retrieve orders for a product or not.

**Reply by reda-alaoui** [reply]
> I currently use JPA projections in combination with JPA entity graphs.
> For example to retrieve a Product object with or without the list of corresponding Orders.

I think your EntityGraph annotation is, at best, just ignored by `spring-data-jpa`.

See:
- https://github.com/spring-projects/spring-data-jpa/issues/1814
- https://github.com/spring-projects/spring-data-jpa/issues/2709

**Reply by z8manu8z** [reply]
It's strange ...
I am simply basing this on the fact that my query using JPA projections does not have the same effect with or without the graph: N+1 queries without, only 1 query with it.
